code/modules/mob/living/simple_animal/hostile/megafauna/megafauna.dm

```diff
@@ -62,4 +67,116 @@
 			adjustBruteLoss(100)
 
 		if(3)
-			adjustBruteLoss(50)
\ No newline at end of file
+			adjustBruteLoss(50)
+
+
+
+/mob/living/simple_animal/hostile/megafauna/proc/grant_achievement(medaltype,scoretype)
+
+	if(medal_type == "Boss")	//Don't award medals if the medal type isn't set
+		return
+
+	for(var/mob/living/L in view(7,src))
+		if(L.stat)
+			continue
+		if(L.client)
+			var/client/C = L.client
+			var/suffixm = BOSS_KILL_MEDAL
+			UnlockMedal("Boss [suffixm]",C)
+			UnlockMedal("[medaltype] [suffixm]",C)
+			SetScore(BOSS_SCORE,C,1)
+			SetScore(score_type,C,1)
+
+/proc/UnlockMedal(medal,client/player)
+
+	if(!player || !medal)
+		return
+	if(global.medal_hub && global.medal_pass && global.medals_enabled)
+		spawn()
+			var/result = world.SetMedal(medal, player, global.medal_hub, global.medal_pass)
+			if(isnull(result))
+				global.medals_enabled = FALSE
+				log_game("MEDAL ERROR: Could not contact hub to award medal:[medal] player:[player.ckey]")
+				message_admins("Error! Failed to contact hub to award [medal] medal to [player.ckey]!")
```

Comment:
Make this a proc, database has a proc around connecting to the database, so this should be one too.